Utilities/StorageFactory/src/RemoteFile.cc

```diff
@@ -52,27 +54,30 @@ int RemoteFile::local(const std::string &tmpdir, std::string &temp) {
   // This is better for grid jobs as the current directory is
   // likely to have more space, and is more optimised for
   // large files, and is cleaned up after the job.
-  if (tmpdir.empty() || tmpdir == ".") {
-    size_t len = pathconf(".", _PC_PATH_MAX);
-    char *buf = (char *)malloc(len);
-    getcwd(buf, len);
+  std::vector<char> temp_chars;
 
-    temp.reserve(len + 32);
-    temp = buf;
-    free(buf);
+  if (tmpdir.empty() || tmpdir == ".") {
+    std::filesystem::path current_path = std::filesystem::current_path();
+    temp_chars.assign(current_path.string().begin(), current_path.string().end());
   } else {
-    temp.reserve(tmpdir.size() + 32);
-    temp = tmpdir;
+    temp_chars.assign(tmpdir.begin(), tmpdir.end());
   }
-  if (temp[temp.size() - 1] != '/')
-    temp += '/';
 
-  temp += "storage-factory-local-XXXXXX";
-  temp.c_str();  // null terminate for mkstemp
+  if (temp_chars.back() != '/')
+    temp_chars.push_back('/');
+
+  std::string suffix = "storage-factory-local-XXXXXX";
+  temp_chars.insert(temp_chars.end(), suffix.begin(), suffix.end());
 
-  int fd = mkstemp(&temp[0]);
+  // Ensure the vector is null-terminated
+  temp_chars.push_back('\0');
+
+  int fd = mkstemp(temp_chars.data());
   if (fd == -1)
-    throwStorageError("RemoteFile", "Calling RemoteFile::local()", "mkstemp()", errno);
+    throw std::system_error(errno, std::generic_category(), "mkstemp failed");
```

Comment:
Please use a `throwStorageError` as that used a cms specific exception type and adds necessary additional information to help diagnose problems.